### DISCUSSION TITLE
Tumblr sign-in gate

### DISCUSSION THREAD

**Discussion by wavygecko** [DandelionSprout/adfilt#642] [Q&A]
I'm trying to figure out how to bypass Tumblr's sign-in gate redirect URLs, but am running into a strange problem.

Example page: `https://www.tumblr.com/explore/trending`

I'm using these filters:

```
||assets.tumblr.com/pop/js/modern/async-login-wall-*.js$script,1p
tumblr.com##+js(ra, data-login-wall-type, , stay)
```

Links with the attribute `data-login-wall-type` are blocked by the sign-in gate, so I'm trying to remove them. (Removing them in the DOM makes the links work correctly.)

However, for some reason with the filter I wrote, the attribute `data-login-wall-type` is not being removed from links. Did I write the filter incorrectly?

**Comment by DandelionSprout** [comment]
After around 15min of me fumbling around, as I myself am far from an expert at `+js` either, these entries *seem* to work:
```
tumblr.com##+js(ra, data-login-wall-type)
tumblr.com##+js(ra, data-login-wall-tag-label)
tumblr.com##+js(ra, data-login-wall-destination)
tumblr.com##+js(ra, data-login-wall-redirect)
tumblr.com##+js(ra, data-login-wall-blog-name)
```

**Reply by wavygecko** [reply]
Thank you for trying that out! Unfortunately those filters don't appear to work for me. :( I can't tell if Tumblr is reinserting those attributes or if there's some kind of race condition that's preventing uBO from properly removing them.

For example, if you click on a username above a post (see below), the URL changes from `https://www.tumblr.com/explore/trending?source=homepage_explore` to `https://www.tumblr.com/explore/trending?redirect_to=%2Fdashboard&source=login_wall`, blocking you from viewing that user's page. However, if you remove only the attribute `data-login-wall-type` from that anchor element then try clicking the it again, the URL properly updates to `https://www.tumblr.com/blog/view/{username}`.

<img width="885" alt="image" src="https://user-images.githubusercontent.com/110956608/187254315-a27d9432-b8a3-4acb-a844-cb8b18a08e6a.png">

So, I'm able to defuse the sign-in gate manually by manipulating the DOM, but for some reason the `ra.js` uBO filters aren't able to do it for me.

**Reply by wavygecko** [reply]
Maybe there's some kind of `onclick` defusion that would work better?

**Comment by MasterKia** [comment]
>  for some reason with the filter I wrote, the attribute data-login-wall-type is not being removed from links. Did I write the filter incorrectly?

This doesn't work?
`tumblr.com##+js(ra, data-login-wall-type, *, stay)`

**Reply by wavygecko** [reply]
It does not. I have no idea why :(

**Reply by MasterKia** [reply]
Also test using Chromium just to make sure.

**Reply by wavygecko** [reply]
What in the world... It works in Chromium but not in Firefox.

**Reply by MasterKia** [reply]
Maybe this is the case:

The problem with Firefox:
https://github.com/uBlockOrigin/uAssets/issues/14204#issuecomment-1202228451

The solution (install Scriptlet Doctor):
https://github.com/uBlockOrigin/uAssets/issues/14456#issuecomment-1221327769

**Reply by wavygecko** [reply]
That sounds like it's probably the issue. Very annoying. 😩 
Thanks for your help!

**Reply by MasterKia** [reply]
If you want to go with this solution, make sure to disable the "limited domain" option in Scriptlet Doctor's settings or just add tumblr to the domains.

**Reply by MasterKia** [reply]
Also if your filters prevent forced sign-in redirection, then maybe you can open an issue/PR for uBO Annoyances list.

**Reply by wavygecko** [reply]
https://github.com/uBlockOrigin/uAssets/issues/14606
